woocommerce/woocommerce
# Explicitly declare PHP version requirements and platform defaults

### All Submissions:


### Changes proposed in this Pull Request:

When running `composer update` on anything more recent than PHP 7.0, doctrine/instantiator version 1.2.0 is installed and tracked in the `composer.lock` file. Unfortunately, this causes some rather cryptic errors when running the WooCommerce test suite in PHP 7.0, as [doctrine/instantiator version 1.1.0 dropped support for PHP < 7.1](https://github.com/doctrine/instantiator/releases/tag/1.1.0). [An example of such an error is available here](https://travis-ci.org/liquidweb/woocommerce-custom-orders-table/jobs/555905558#L689].

This commit adds two things to the `composer.json` file:

1. Explicit platform requirements (`"php": "^5.6|^7.0"`)
2. A `config.platform` value, which essentially tells Composer "when you're installing/updating dependencies, assume it's on a machine running PHP 7.0".

My assumption is the current `composer.lock` file was generated in an environment running PHP 7.0, which is why current Travis runs are working but running `composer update` on anything higher than PHP 7.0 causes the tests to start failing within the PHP 7.0 test runner.

### Alternate approach

If we don't want to pin to an unsupported PHP version, we could alternately [download the PHPUnit 6 PHAR file](https://phpunit.de/getting-started/phpunit-6.html) for PHP 7.0 tests rather than relying on a Composer installation; in my testing, the same copy of the codebase being run through the PHAR (as opposed to `vendor/bin/phpunit`) produced different results (both running PHPUnit 6.5.14):

```sh
# No errors
$ php7.0 phpunit6.phar

# Errors as seen in the linked Travis job
$ php7.0 vendor/bin/phpunit
```
### How to test the changes in this Pull Request:

1. In a PHP > 7.0 environment, run `composer update`
2. Verify that the version of doctrine/instantiator installed is 1.0.5
3. Using the dependencies installed in a higher environment, run PHPUnit using PHP 7.0.

### Other information:


### Changelog entry

> Explicitly pin the target platform for Composer to PHP 7.0.
